spec/consensus/light-client.md

```diff
@@ -316,3 +316,27 @@ func Backwards(h1,h2) bool {
   return (hash(h2) == old.Header.hash)
  }
 ```
+
+### Failure Detection Module
+
+The light client must be able to detect failures in the trust assumptions.
+Ideally, it should never trust a header, which was forged by malicious
+validators.
+
+To meet the above requirement, it **must be connected to at least one honest
+full node**.
+
+In practice, this means connecting to one or more geographically distributed
+full nodes (called witness), which belong to different companies.
+
+After the light client verifies a new header (`H`), it should cross-check `H`
+with the headers from all witnesses. Cross-checking means comparing hashes of
+the headers. If any two hashes (or more) diverge, there's a fork.
+
+And since there is no way for the light client to detect who's lying to it
+(which full node - primary or one of the witnesses), it must form an evidence
+and send it to all connected full nodes. The evidence will typically contain
+a set of diverged headers (including commits for each of them).
```

Comment:
I guess there might be cases where the lite client may detect who is lying, e.g., if basic verification fails (e.g., the header is signed by nodes that are not in the validator set).

Reply:
Not if we're running bisection (which is most common use-case).